Basic Agent Usage -- merge with Agent main page

iadjivon opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?

This PR merges the content from Basic Agent usage into the Agent main page. It also:
- moves over the aliases from Basic Agent usage
- adds the Basic Agent usage page as an alias in the Agent main page
- comments out the Basic Agent section in the main menu file (should this be removed entirely?)



### Merge instructions

<!-- 
If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. 
If the PR is ready to be merged once it receives the required reviews, check the box below after you've created the PR.
-->

Merge readiness:
- [ ] Ready for merge

**For Datadog employees**:

Your branch name MUST follow the `<name>/<description>` convention and include the forward slash (`/`). Without this format, your pull request will not pass CI, the GitLab pipeline will not run, and you won't get a branch preview. Getting a branch preview makes it easier for us to check any issues with your PR, such as broken links.

If your branch doesn't follow this format, rename it or create a new branch and PR.

[6/5/2025] Merge queue has been disabled on the documentation repo. If you have write access to the repo, the PR has been reviewed by a Documentation team member, and all of the required checks have passed, you can use the **Squash and Merge** button to merge the PR. If you don't have write access, or you need help, reach out in the #documentation channel in Slack.

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

brett0000FF commented on config/_default/menus/main.en.yaml:
```diff
@@ -304,66 +304,66 @@ menu:
-      url: agent/basic_agent_usage/source/
-      weight: 120
-      parent: basic_agent_usage
+    # - name: Basic Agent Usage
```
Per your question, yes these should just be deleted.

brett0000FF commented on content/en/agent/basic_agent_usage/_index.md:
```diff
@@ -1,6 +1,8 @@
 ---
```
I believe this page should be removed entirely? No need to keep it if we want users to go to the main agent page instead. Leaving it also creates some conflicting aliases. For example `/guides/basic_agent_usage/` is an alias to this page as well as on the main Agent landing page.

What do you think about removing this folder entirely and putting the child pages under a new folder with Supported Platforms as the index? Note: You'd have to add some additional aliases for moving those child pages, but I think ultimately that would be cleaner. We could also put these child pages under Supported Platforms in the nav too; otherwise, those are no longer accessible.

brett0000FF commented on content/en/agent/guide/how-do-i-uninstall-the-agent.md:
```diff
@@ -4,19 +4,19 @@ description: "Learn how to uninstall the Datadog Agent from different operating
 ---
 
-For information on uninstalling the Agent on your system, consult the [Basic Agent Usage][1] page for your operating system, platform, or configuration tool.
+For information on uninstalling the Agent on your system, consult the [Supported Platforms][1] page for your operating system, platform, or configuration tool.
```
It doesn't seem like the Supported Platform page reliably links to the pages with uninstall instructions. I think the most straightforward approach would be to refer to the the links on _this_ page that are populated by the `uninstall-agent` partial.

iadjivon replied:
Is it just me or is the `uninstall-agent` partial missing? I do not see it under the partials section, but may be missing something. Thanks!

brett0000FF reviewed (changes requested):
Looks good! Main things that I noticed:

- Yes, the commented out nav entries should be deleted entirely.
- However, to that point, I don't think we should remove the child pages from the nav (e.g., AIX, Linux, etc.)? Seems like they are important, so we could put them under Supported Platforms?
- `content/en/agent/basic_agent_usage/_index.md` -> This page should be removed since it's just a duplicate now. Related to the previous bullet and this point, I think we could remove this folder entirely, and move the child pages into a new Supported Technologies folder. 

Happy to chat through things if you need any clarification! Should just need a few tweaks and this'll be all set. 🚀

iadjivon commented:
Hi Brett, thanks for this feedback! I completed the following:

**NAV**
- removed the nav entries for the basic agent usage main page and `/agent/basic_agent_usage/source/` pages

**FILES**
- moved the child pages to be under `supported platforms`
- aliases all of the moved pages in the frontmatter of the  `supported_platform > _index.md` file 
- `/agent/basic_agent_usage/source/` is deleted

brett0000FF commented on content/en/agent/basic_agent_usage/source.md:
I might have missed it, but why did we delete this file? It seems to contain info that doesn't exist elsewhere.

iadjivon replied:
Yup! this was a mistake. reverted it!

brett0000FF commented on content/en/agent/supported_platforms/ansible.md:
```diff
@@ -0,0 +1,103 @@
+---
```
Re: Previous comment. Files like this need an alias to the old path under basic_agent_usage.

brett0000FF commented on content/en/agent/guide/how-do-i-uninstall-the-agent.md:
```diff
@@ -4,19 +4,19 @@ description: "Learn how to uninstall the Datadog Agent from different operating
-For information on uninstalling the Agent on your system, consult the [Basic Agent Usage][1] page for your operating system, platform, or configuration tool.
+For information on uninstalling the Agent on your system, select your operating system, platform, or configuration tool.
 
 {{< uninstall-agent >}}
```
This is in https://github.com/DataDog/documentation/blob/master/layouts/shortcodes/uninstall-agent.html not the partial folder FYI (addressing your question).

It is hardcoded to get child pages under `agent/basic_agent_usage`. You'll have to update the shortcode to get child pages from `agent/supported_platforms`.

For example, the uninstall-agent file should be updated to look like this:

```
<ul>
{{ range (site.GetPage "agent/supported_platforms").Pages }}
  <li><a href="{{ .RelPermalink }}">{{ replace .Title "Basic Agent Usage for" "" }}</a></li>
{{ end }}
</ul>
```

iadjivon commented on content/en/agent/_index.md:
```diff
@@ -22,6 +22,22 @@ aliases:
+  - /agent/basic_agent_usage/sccm/
+  - /agent/basic_agent_usage/windows/
+  - /agent/basic_agent_usage/chef/
+  - /agent/basic_agent_usage/source/
```
Oh duh! That makes complete sense! 🤦🏾‍♀️

brett0000FF commented:
Preview with cache turned off in cloned PR: https://docs-staging.datadoghq.com/brett.blue/no-cache-basic-agent/agent/supported_platforms/ansible/

iadjivon commented on content/en/agent/supported_platforms/aix.md:
```diff
@@ -2,7 +2,7 @@
 title: Basic Agent Usage for AIX
```
```suggestion
title: AIX
```

iadjivon commented on config/_default/menus/main.en.yaml:
```diff
@@ -304,81 +304,76 @@ menu:
+      url: agent/supported_platforms/heroku/
+      weight: 405
+      parent: agent_supported_platforms
     - name: Mac OS X
```
```suggestion
    - name: MacOS 
```

jszwedko commented on content/en/agent/_index.md:
```diff
@@ -43,27 +47,128 @@ Agent v7 is available. <a href="/agent/versions/upgrade_to_agent_v7">Upgrade to
 
 The Datadog Agent is software that runs on your hosts. It collects events and metrics from hosts and sends them to Datadog, where you can analyze your monitoring and performance data. The Datadog Agent is open source and its source code is available on GitHub at [DataDog/datadog-agent][1].
 
-**It is recommended to fully install the Agent.** However, a standalone DogStatsD package is available for Amazon Linux, CentOS, Debian, Fedora, Red Hat, SUSE, and Ubuntu. This package is used in containerized environments where DogStatsD runs as a sidecar or environments running a DogStatsD server without full Agent functionality.
```
@brett0000FF we seem to have lost this content regarding the standalone DogStatsD package. Could we revive it? I realized when I went to point someone to it and discovered it was missing.

iadjivon replied:
Hi @jszwedko Thanks for this note and thank you @brett0000FF for looping me in! 

I worked with @ethandebnath7702 on this and have looped him in to comment on next steps here!
